Libraries/HyperV.psm1

```diff
@@ -258,13 +272,39 @@ Function DeleteHyperVGroup([string]$HyperVGroupName, [string]$HyperVHost) {
         LogMsg "Stop-VM -Name $($_.Name) -Force -TurnOff "
         $vm = $_
         Stop-VM -Name $vm.Name -Force -TurnOff -ComputerName $HyperVHost
+        Remove-VMSnapshot -VMName $vm.Name -ComputerName $HyperVHost `
+            -IncludeAllChildCheckpoints -Confirm:$false
+        if (!$?) {
+            LogErr ("Failed to remove snapshots for VM {0}" -f @($vm.Name))
+            return $false
+        }
+        Wait-VMStatus -VMName $vm.Name -VMStatus "Operating Normally" -RetryInterval 2 `
+            -HvServer $HyperVHost
+        $vm = Get-VM -Name $vm.Name -ComputerName $HyperVHost
         $vm.HardDrives | ForEach-Object {
-            $vhd = $_
-            if (Test-Path -Path $vhd.Path) {
-                Invoke-Command -ComputerName $HyperVHost -ScriptBlock {
-                    Remove-Item -Path $args[0] -Force } -ArgumentList $vhd.Path
-                LogMsg "VHD $($vhd.Path) removed!"
+            $vhdPath = $_.Path
+            $invokeCommandParams = @{
+                "ScriptBlock" = {
+                    Remove-Item -Path $args[0] -Force
+                };
+                "ArgumentList" = $vhdPath;
+                "ErrorAction" = "Stop";
+            }
+            if ($HyperVHost -ne "localhost" -and $HyperVHost -ne $(hostname)) {
+                $invokeCommandParams.ComputerName = $HyperVHost
+            }
+            Invoke-Command @invokeCommandParams
+            if (!$?) {
+                LogMsg "Failed to remove ${vhdPath} using Invoke-Command"

```

Comment:
maybe update the text to below, to make it clearer that invoke failed, but there is a retry removal using unc
Failed to remove ${vhdPath} using Invoke-Command, retrying with UNC paths

Reply:
This is part of Adi Vladu's PR with improvements to the vhd cleanup